c/typedb_driver.i

```diff
@@ -116,12 +117,75 @@ struct TransactionCallbackDirector {
 %{
 #include <memory>
 #include <iostream>
+#include <mutex>
 #include <unordered_map>
-static std::unordered_map<size_t, TransactionCallbackDirector*> transactionOnCloseCallbacks {};
```

Comment:
This had a race condition if the driver calls a callback at the same time as a user adding one to a new transaction.